mat/band.go

```diff
@@ -136,6 +136,12 @@ func NewBandDense(r, c, kl, ku int, data []float64) *BandDense {
 	}
 }
 
+// NewDiagonal is a convenience function that returns a diagonal matrix represented by a
+// BandDense. The length of data must be min(r, c) otherwise NewDiagonal will panic.
+func NewDiagonal(r, c int, data []float64) *BandDense {
```

Comment:
Please leave this out of this pull request.

I think we want a `SymBanded` interface / type. It makes a huge difference in `distmv.NewNormal` (O(n^3) to O(n) for computing the Cholesky), and is probably a common case in the spatial statistics world also. Typically, diagonal matrices are also square, so I think we would want this to return a SymBandDense by default.